### ISSUE TITLE
Undefined reference on Windows Subsystem for Linux

### ISSUE THREAD

**Issue by buddha314** [jtv/libpqxx#314]
I'm using CLion (which forces CMake) with the WSL toolstack.

```
cmake_minimum_required(VERSION 3.13)
project(conana)

set(CMAKE_CXX_STANDARD 20)
include(${CMAKE_BINARY_DIR}/cmake-build-debug/conanbuildinfo.cmake)
conan_basic_setup()
set(APPEND CMAKE_PREFIX_PATH "/mnt/c/Users/b/src/libpqxx/")

set(SOURCES main.cpp)

add_executable(conana ${SOURCES})
#target_link_libraries(${project} ${CONAN_LIBS})
```

Produces the error

```
====================[ Build | conana | Debug ]==================================
/usr/bin/cmake --build /mnt/c/Users/b/CLionProjects/conana --target conana -- -j 8
[ 50%] Building CXX object CMakeFiles/conana.dir/main.cpp.o
[100%] Linking CXX executable bin/conana
/usr/bin/ld: CMakeFiles/conana.dir/main.cpp.o: in function `__static_initialization_and_destruction_0(int, int)':
/home/buddha/.conan/data/libpqxx/7.0.1/bincrafters/stable/package/cfd7bc19fdc01c984df6e1fe4e7d6856d5f9b76a/include/pqxx/strconv.hxx:76: undefined reference to `pqxx::internal::demangle_type_name[abi:cxx11](char const*)'
collect2: error: ld returned 1 exit status
make[3]: *** [CMakeFiles/conana.dir/build.make:84: bin/conana] Error 1
make[2]: *** [CMakeFiles/Makefile2:73: CMakeFiles/conana.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:85: CMakeFiles/conana.dir/rule] Error 2
make: *** [Makefile:118: conana] Error 2
```

I appreciate your help.

**Comment by tt4g** [comment]
This repository is not maintained conan recipe, it is recommended move issue to [conan\-io/conan\-center\-index](https://github.com/conan-io/conan-center-index) repository.

**Comment by buddha314** [comment]
@tt4g Thank you for the link library suggestion, when I uncomment it, I get

```
====================[ Build | conana | Debug ]==================================
/usr/bin/cmake --build /mnt/c/Users/b/CLionProjects/conana --target conana -- -j 8
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- Current conanbuildinfo.cmake directory: /mnt/c/Users/b/CLionProjects/conana/cmake-build-debug
-- Conan: Compiler GCC>=5, checking major version 9
-- Conan: Checking correct version: 9
CMake Error at CMakeLists.txt:15 (target_link_libraries):
  Cannot specify link libraries for target "pqxx" which is not built by this
  project.


-- Configuring incomplete, errors occurred!
See also "/mnt/c/Users/b/CLionProjects/conana/CMakeFiles/CMakeOutput.log".
make: *** [Makefile:176: cmake_check_build_system] Error 1
```

I do not think the issue is related to conan so I had this behavior before and I thought I could solve it with package management.

**Comment by tt4g** [comment]
Look at the following error message "Cannot specify link libraries for target "pqxx" which is not built by this project."
CMake is reporting this error message because it is unable to find the CMake target pqxx.
If you built pqxx with conan, then this is the conan problem.

**Comment by tt4g** [comment]
If conan can't find the CMake package you built, it may be because the environment in which you built the CMake package (e.g. compiler, environment variables) is different from the environment in which you build your project.

There is an official CMake and conan integration module that can easily work around these problems: https://github.com/conan-io/cmake-conan
Can you try using this module and see if it resolves the error?

**Comment by buddha314** [comment]
Thanks for you advice.  It was not conan, but I was able get past that error with this CMake

```
cmake_minimum_required(VERSION 3.13)
project(conana)

set(CMAKE_CXX_STANDARD 20)

set(PQXX /usr/local/include/pqxx)
find_library(PQXX_LIB pqxx)

set(SOURCES main.cpp)

add_executable(conana ${SOURCES})
TARGET_LINK_LIBRARIES(conana ${PQXX_LIB} ${PQ_LIB})
```
I think the `find_library` was the key, but cmake isn't my strong suit.

And I just saw your post, i will try that as well.  Again, thanks for the help.

**Comment by tt4g** [comment]
@buddha314 Are you using Conan Recipe `libpqxx/7.0.1@bincrafters/stable`?
This recipe is no longer being maintained.
Older recipes may cause problems with CMake target definition.
I recommend that you use `libpqxx/7.0.7` instead.

**Comment by buddha314** [comment]
Thank you, I can change that and give it a shot.  I'm on to another error when I try to connect which I'll dive into before I bug you again.

**Comment by buddha314** [comment]
I got a different undefined reference error but here is the one that is working now, in case someone needs it.

I downloaded and built `libpqxx` into my `b/src/libpqxx` directory as per the directions on the home page.  Then my `CMakeLists.txt` looks like

```
cmake_minimum_required(VERSION 3.13)
project(conana)

set(CMAKE_CXX_STANDARD 20)

find_library(PQXX_LIB pqxx REQUIRED)
find_library(PQ_LIB pq REQUIRED)

set(SOURCES main.cpp)

add_executable(conana ${SOURCES})
TARGET_LINK_LIBRARIES(conana ${PQXX_LIB} ${PQ_LIB})
```

It seems the two `find_library` commands were the key.  I ended up bypassing conan, but i'm going to try to mix that back in.

Much obliged, tt4g.
